client/verta/tests/unit_tests/test_runtime.py

```diff
@@ -0,0 +1,205 @@
+            # time.sleep(random.uniform(0, 1))
+        assert hasattr(runtime._THREAD, 'logs') == False
+        assert hasattr(runtime._THREAD, 'validate') == False
+        assert test_ctx.logs() == {'test_key': value}
+
+    with futures.ThreadPoolExecutor(max_workers=10) as executor:
+        list(
+            executor.map(
+                log_in_context,
+                [{f'thread_{x}_key': f'thread_{x}_val'} for x in range(50)]
+            )
+        )
+
+
+def test_init_thread_logs():
+    """
+    An empty dict is initialized by default.
+    """
+    assert runtime._get_thread_logs() == {}
+
+
+def test_set_thread_logs():
+    runtime._set_thread_logs({'test_set': 'test_set'})
+    assert runtime._THREAD.logs ==  {'test_set': 'test_set'}
+
+
+def test_get_thread_logs():
+    runtime._set_thread_logs({'test_get': 'test_get'})
+    assert runtime._get_thread_logs() ==  {'test_get': 'test_get'}
+
+
+def test_init_thread_validate():
+    """
+    A boolean defaulting to False is initialized by default.
+    """
+    assert runtime._get_validate_flag() == False
+
+
+def test_set_thread_validate():
+    runtime._set_validate_flag(True)
+    assert runtime._THREAD.validate == True
+
+
+def test_get_thread_validate():
+    runtime._set_validate_flag(True)
+    assert runtime._get_validate_flag() == True
+    runtime._set_validate_flag(False)
+
+
+def test_logs_are_clean_on_entry():
+    """ Thread local var for logs is a blank dict when entering context. """
+    with runtime.context() as ctx:
+        assert ctx.logs() == {}
+
+
+def test_logs_are_clean_on_exit():
+    """ Thread local var for logs is a blank dict after exiting context. """
+    with runtime.context() as ctx:
+        runtime.log('fake_log', 'fake_value')
+    assert runtime._get_thread_logs() == {}  # outside the context
+
+
+def test_final_log_entry_instance_variable():
+    """ The instance of the context manager maintains the final log entry in logs(). """
+    with runtime.context() as ctx:
+        runtime.log('fake_log', 'fake_value')
+    assert ctx.logs() == {'fake_log': 'fake_value'}  # outside the context
+
+
+def test_validate_flag_true():
+    """ The validate argument triggers json validation as expected. """
+    with runtime.context(validate=True):
+        with pytest.raises(TypeError):
+            runtime.log('obviously_not_jsonable', unittest.TestCase)
+
+
+def test_validate_flag_false():
+    """ The validate argument defaults to False and does not trigger json validation. """
+    with runtime.context():
+        runtime.log('obviously_not_jsonable', unittest.TestCase)
+
+
+def test_validate_reset_on_exit():
+    """ Thread local var for validate is reset to default (False) after exiting context. """
+    with runtime.context(validate=True):
+        runtime.log('test', {'test_key': 'test_val'})
+        assert runtime._get_validate_flag() == True  # inside the context
+    assert runtime._get_validate_flag() == False  # outside the context
```

Comment:
`context.__exit__()` deletes `_THREAD.validate`, but here `runtime._get_validate_flag()` sets `_THREAD.validate = False`. It seems like the implementation and this assertion are at odds with each other.

What is the state that we want when a context is exited? That `_THREAD.validate` doesn't exist, or that `_THREAD.validate` is `False`?